From 7ac6308af5ea6b4583930da73a2c184e0bbccc98 Mon Sep 17 00:00:00 2001
From: siddhesh <siddhesh>
Date: Thu, 27 Sep 2012 08:57:14 +0000
Subject: [PATCH 1/1] 	* gdbtypes.c (lookup_array_range_type): Expand parameters
 	LOW_BOUND and HIGH_BOUND to LONGEST.
 	(lookup_string_range_type): Likewise.
 	* gdbtypes.h (lookup_array_range_type): Likewise.
 	(lookup_string_range_type): Likewise.
 	* valops.c (value_cstring): Expand parameter LEN to ssize_t.
 	Expand HIGHBOUND to ssize_t.
 	(value_string): Likewise.
 	* value.h (value_cstring): Expand parameter LEN to ssize_t.
 	(value_string): Likewise.

---
 gdb/ChangeLog  |   13 +++++++++++++
 gdb/gdbtypes.c |    4 ++--
 gdb/gdbtypes.h |    4 ++--
 gdb/valops.c   |    8 ++++----
 gdb/value.h    |    4 ++--
 5 files changed, 23 insertions(+), 10 deletions(-)

Index: gdb-7.2/gdb/gdbtypes.c
===================================================================
--- gdb-7.2.orig/gdb/gdbtypes.c
+++ gdb-7.2/gdb/gdbtypes.c
@@ -918,7 +918,7 @@ create_array_type (struct type *result_t
 
 struct type *
 lookup_array_range_type (struct type *element_type,
-			 int low_bound, int high_bound)
+			 LONGEST low_bound, LONGEST high_bound)
 {
   struct gdbarch *gdbarch = get_type_arch (element_type);
   struct type *index_type = builtin_type (gdbarch)->builtin_int;
@@ -954,7 +954,7 @@ create_string_type (struct type *result_
 
 struct type *
 lookup_string_range_type (struct type *string_char_type,
-			  int low_bound, int high_bound)
+			  LONGEST low_bound, LONGEST high_bound)
 {
   struct type *result_type;
 
Index: gdb-7.2/gdb/gdbtypes.h
===================================================================
--- gdb-7.2.orig/gdb/gdbtypes.h
+++ gdb-7.2/gdb/gdbtypes.h
@@ -1478,7 +1478,7 @@ extern struct type *create_range_type (s
 
 extern struct type *create_array_type (struct type *, struct type *,
 				       struct type *);
-extern struct type *lookup_array_range_type (struct type *, int, int);
+extern struct type *lookup_array_range_type (struct type *, LONGEST, LONGEST);
 
 extern CORE_ADDR type_range_any_field_internal (struct type *range_type,
 						int fieldno);
@@ -1494,7 +1494,7 @@ extern void finalize_type (struct type *
 
 extern struct type *create_string_type (struct type *, struct type *,
 					struct type *);
-extern struct type *lookup_string_range_type (struct type *, int, int);
+extern struct type *lookup_string_range_type (struct type *, LONGEST, LONGEST);
 
 extern struct type *create_set_type (struct type *, struct type *);
 
Index: gdb-7.2/gdb/valops.c
===================================================================
--- gdb-7.2.orig/gdb/valops.c
+++ gdb-7.2/gdb/valops.c
@@ -1777,11 +1777,11 @@ value_array (int lowbound, int highbound
 }
 
 struct value *
-value_cstring (char *ptr, int len, struct type *char_type)
+value_cstring (char *ptr, ssize_t len, struct type *char_type)
 {
   struct value *val;
   int lowbound = current_language->string_lower_bound;
-  int highbound = len / TYPE_LENGTH (char_type);
+  ssize_t highbound = len / TYPE_LENGTH (char_type);
   struct type *stringtype
     = lookup_array_range_type (char_type, lowbound, highbound + lowbound - 1);
 
@@ -1800,11 +1800,11 @@ value_cstring (char *ptr, int len, struc
    string may contain embedded null bytes.  */
 
 struct value *
-value_string (char *ptr, int len, struct type *char_type)
+value_string (char *ptr, ssize_t len, struct type *char_type)
 {
   struct value *val;
   int lowbound = current_language->string_lower_bound;
-  int highbound = len / TYPE_LENGTH (char_type);
+  ssize_t highbound = len / TYPE_LENGTH (char_type);
   struct type *stringtype
     = lookup_string_range_type (char_type, lowbound, highbound + lowbound - 1);
 
Index: gdb-7.2/gdb/value.h
===================================================================
--- gdb-7.2.orig/gdb/value.h
+++ gdb-7.2/gdb/value.h
@@ -417,9 +417,9 @@ extern struct value *value_mark (void);
 
 extern void value_free_to_mark (struct value *mark);
 
-extern struct value *value_cstring (char *ptr, int len,
+extern struct value *value_cstring (char *ptr, ssize_t len,
 				    struct type *char_type);
-extern struct value *value_string (char *ptr, int len,
+extern struct value *value_string (char *ptr, ssize_t len,
 				   struct type *char_type);
 extern struct value *value_bitstring (char *ptr, int len,
 				      struct type *index_type);
